perceval/backends/mediawiki.py

```diff
@@ -0,0 +1,475 @@
+
+        :returns: a generator of pages
+        """
+
+        logger.info("Looking for pages at url '%s'", self.url)
+
+        self._purge_cache_queue()
+        npages = 0  # number of pages processed
+
+        # The pages are organized in content namespaces
+        raw_namespaces = self.client.get_namespaces()
+        namespaces = json.loads(raw_namespaces)["query"]["namespaces"]
+        namespaces_contents = [ns for ns in namespaces if 'content' in namespaces[ns].keys()]
+        self._push_cache_queue(raw_namespaces)
+        self._flush_cache_queue()
+
+        arvcontinue = '' # pagination for getting revisions and their pages
+        while arvcontinue is not None:
+            raw_pages = self.client.get_pages_from_allrevisions(namespaces_contents, from_date, arvcontinue)
+            self._push_cache_queue(raw_pages)
+            self._flush_cache_queue()
+            data_json = json.loads(raw_pages)
+            if 'continue' in data_json:
+                arvcontinue = data_json['continue']['arvcontinue']
+            else:
+                arvcontinue = None
+            pages_json = data_json['query']['allrevisions']
+            for page in pages_json:
+                urltitle = urllib.parse.urlencode({'titles':page['title']})
+                revisions_raw = self.client.get_revisions(urltitle)
+                self._push_cache_queue(revisions_raw)
+                self._flush_cache_queue()
+                page_reviews = self.__build_page_reviews(page, json.loads(revisions_raw))
+                yield page_reviews
+                npages += 1
+
+        logger.info("Total number of pages: %i", npages)
+
+
+
+    def __fetch_pre1_27(self, from_date=None):
+        """Fetch the pages from the backend url.
+
+        The method retrieves, from a MediaWiki url, the
+        wiki pages.
+
+        :returns: a generator of pages
+        """
+
+        def get_page_reviews(title):
+            urltitle = urllib.parse.urlencode({'titles':title})
+            revisions_raw = self.client.get_revisions(urltitle)
+            self._push_cache_queue(revisions_raw)
+            self._flush_cache_queue()
+            page_reviews = self.__build_page_reviews(page, json.loads(revisions_raw))
+            return page_reviews
+
+        logger.info("Looking for pages at url '%s'", self.url)
+
+        # from_date can not be older than MAX_RECENT_DAYS days ago
+        if from_date:
+            if (datetime.datetime.utcnow()-from_date).days >= MAX_RECENT_DAYS:
+                cause = "Can't get incremental pages older than %i days." % MAX_RECENT_DAYS
+                cause += " Do a complete analysis without from_date for older changes."
+                raise BackendError(cause=cause)
+
+        self._purge_cache_queue()
+        nns = 0 # number of namespaces
+        npages = 0  # number of pages processed
+
+        # The pages are organized in content namespaces
+        raw_namespaces = self.client.get_namespaces()
```

Comment:
I have changed the code and I can find the similar code. I have thought during code of sharing more code between implementations but there are several things that make implementations different. But we can try to refactor it more. Let's review it after this first iteration.